### PR TITLE
4.x Refactor RouteGroup & Route

### LINKED ISSUE #2598: Why is RouteGroup a Routable when it can't be routed to?
This probably needs tidying up.

### PR DESCRIPTION
**Pull request by l0gicgate**
This is pull 1 out of 7 to complete the goals set in #2604 

- Refactor `RouteGroup` so it doesn't extend `Routable`
- Implement `appendMiddlewareToDispatcher()` method to avoid encapsulation leakage
- Refactor `Route::finalize()` and fold it into `Route::run()`
- Remove `Routable` as it is now obsolete.

This PR effectively closes #2598, #2599 and #2600.

- [x] Discussion
- [x] Write Tests

### CODE REVIEW

File: Slim/Route.php [hunk 1]
```diff
@@ -312,61 +313,29 @@ public function prepare(ServerRequestInterface $request, array $arguments): Rout
...
-     */
-    public function finalize(): void
-    {
-        if ($this->finalized) {
-            return;
-        }
-
-        // Chain a new (outer) middleware dispatcher for route-group-middleware
-        // to the existing route-middleware dispatcher
+        // Add Middleware From Groups
```
Comment (akrabat) [inline Slim/Route.php]: Lower case `f` and `g`

**Comment by akrabat** [comment]
Looks reasonable to me.

File: Slim/Route.php (hunk 1 above)
Comment (l0gicgate) [reply Slim/Route.php]: Fixed

**Review by akrabat — APPROVED** [review]

**Comment by akrabat** [comment]
Do we need a CHANGELOG entry?

**Comment by l0gicgate** [comment]
@akrabat yeah I think so. The public interface has changed for both `Route` and `RouteGroup` so I'll add an entry.

**Pull request merged**